### PR TITLE
Expand SearchTable component to search and filter by topics and type

### PR DESCRIPTION
**Pull request by gabrieltanchen**
#### Description of changes
resolves CIGIHub/cigi-tickets#248
resolves CIGIHub/cigi-tickets#204

Added the search bar to the multimedia and publications search tables, which updates the AJAX url to add search and filtering query params.

I implemented an API endpoint for `/topics` to fill in the topics dropdown.

### CODE REVIEW

**Comment by gabrieltanchen** [comment]
This also resolves CIGIHub/cigi-tickets#204

**Comment by johnbxu** [comment]
Not sure if this should be its own ticket, but on live, no results shows this: 
image

File: cigionline/static/js/components/SearchTable.js [hunk 1]
```diff
@@ -47,12 +110,83 @@ class SearchTable extends React.Component {
       });
   }
 
+  getTopics() {
+    fetch(encodeURI('/api/topics/?limit=50&offset=0&fields=title'))
```
Comment (johnbxu) [inline cigionline/static/js/components/SearchTable.js:114]: I know there aren't 50 topics, but is there a need for `?limit=50` here?

**Comment by gabrieltanchen** [comment]
@johnbxu Updated with empty state. I also lowered the topics query limit to 40. There are currently 31 topics so I'm leaving a little room for more to be added. If we don't add the limit to the query, the Wagtail API defaults to 20 results.


Screen Shot 2020-12-15 at 12 13 05 PM

**Review by johnbxu — APPROVED** [review]

**Pull request merged**